code/rounds/InProgressRound.cs

```diff
@@ -41,6 +44,36 @@ public override void OnPlayerLeave(TTTPlayer player)
             }
         }
 
+        protected override void OnStart()
+        {
+            if (Host.IsServer)
+            {
+                foreach (Client client in Client.All)
+                {
+                    if (client.Pawn is not TTTPlayer player)
+                    {
+                        continue;
+                    }
+
+                    if (player.LifeState == LifeState.Dead)
+                    {
+                        player.Respawn();
+                    }
+
+                    if (!Players.Contains(player))
+                    {
+                        AddPlayer(player);
+                    }
+
+                    // TODO: Remove once we can spawn in weapons into the map, for now just give the guns to people.
+                    player.Inventory.DeleteContents();
+                    player.Inventory.Add(new Shotgun(), true);
+                }
+
+                AssignRoles();
```

Comment:
In TTT, we had a "BeginRound" round to calculating stuff etc., maybe this would be good as well to reduce issues with time is not the exact round time because things are getting calculated and similar things?

Reply:
Hmm, could you go into some more depth of how you see this functioning? We have `OnStart()` available as an overwrite for all rounds.

Reply:
More like a `Waiting` -> `Preparing` -> `Begin` -> `InProgress` -> `Post`. Or a better idea maybe would be doing the role loading stuff etc. in an event that is called in the end of `Preparing` and before starting `InProgress` (as a condition)?

Reply:
Didn't add in a new round, but, should have solved any issues with ingame round time.